noxfile.py

```diff
@@ -30,13 +32,16 @@ def tests_impl(
     implementation_name, release_level = session_python_info.split(" ")
 
     # Install deps and the package itself.
-    session.install("-r", "dev-requirements.txt")
-    if len(extras) > 0:
-        session.install(f".[{extras}]")
-    else:
-        session.install(".")
+    session.run_install(
+        "uv",
+        "sync",
```

Comment:
Should we add `--frozen` to avoid touching the lockfile, as mentioned in the nox docs? https://nox.thea.codes/en/stable/cookbook.html#using-a-lockfile

Reply:
Yes, let's add the flag